bot/exts/moderation/infraction/_utils.py

```diff
@@ -280,3 +283,25 @@ async def send_private_embed(user: MemberOrUser, embed: discord.Embed) -> bool:
             "The user either could not be retrieved or probably disabled their DMs."
         )
         return False
+
+
+class StaffBanConfirmationView(interactions.ViewWithUserAndRoleCheck):
+    """The confirmation view that is sent when a moderator attempts to ban a staff member."""
+
+    @discord.ui.button(label="Confirm", style=ButtonStyle.red)
+    async def confirm(self, interaction: Interaction, button: Button) -> None:
+        """Callback coroutine that is called when the "confirm" button is pressed."""
+        await interaction.response.send_message("Applied infraction.")
+        await interaction.message.edit(view=None)
```

Comment:
Ah, this should be getting done by bot-core automatically, but I notice that it only gets done on timeout. I'll update bot-core so that it works on stop too, could you remove this line and the similar one in `cancel()`?

Reply:
@ChrisLovering I've tested this locally with the changes from https://github.com/python-discord/bot-core/pull/184 and it works, save having to change the `.stop` to be asynchronous. Might the change in https://github.com/python-discord/bot-core/pull/184 break this functionality anywhere else in the codebase?

Reply:
Good point. I've just updated that PR to make `stop()` synchronous to avoid this issue.